### DISCUSSION TITLE
Why does `register` not support `defaultValue`?

### DISCUSSION THREAD

**Discussion by timkindberg** [react-hook-form/react-hook-form#8392] [Ideas]
Hello thank you for the awesome library!! We are migrating from Formik.

We are wondering though, why doesn't register support defaultValue? I am really loving defining all of my fields in their own isolated components, with validation logic and initial values all co-located. I'd rather not specify the defaultValue at the form level.

I've found that if I declare register in conjunction with resetField I can achieve a `defaultValue`. Like this:

```tsx
  const field = React.useMemo(() => {
    const field = register("firstName");
    resetField("firstName", { defaultValue: "Tim" });
    return field;
  }, []);
```

Code Sandbox:
https://codesandbox.io/s/useformstate-forked-t7b0gq?file=/src/App.js:472-657

Is there a problem with this approach? Or could the same thing be done internally to add support for `defaultValue` to `register`?

In general I'm also wondering why not balance the APIs of `register` and `useController`. It seems it could be achieved and then have more predictable usage. Saw a discussion about this here: https://github.com/react-hook-form/react-hook-form/discussions/4542

Update: It even works when the registered field is watched, it has the defaultValue. But only if the watched field is mounted first. Is this restriction the reason that `defaultValue` was removed from register?

**Comment by bluebill1049** [comment]
because defaultValue belongs to inputs eg `<input defaultValue="test" />` and register already capture a lot props and prefer to leave those native attributes to the input itself, if you have issue with that you can always provide default value at `useForm`

**Reply by timkindberg** [reply]
Is it technically achievable to allow the defaultValue to be set at the field level? I'm not talking just about passing `defaultValue` to the input, but more like how it works if you pass to `useForm`. The reason I ask is because I really like decoupling my fields from the form and declare them in isolation. Thank you for the quick response!!
